src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -0,0 +1,193 @@
+package edu.ksu.canvas.attendance.submitter;
+
+
+import edu.ksu.canvas.attendance.entity.AttendanceAssignment;
+import edu.ksu.canvas.attendance.entity.AttendanceCourse;
+import edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException;
+import edu.ksu.canvas.attendance.model.AttendanceSummaryModel;
+import edu.ksu.canvas.attendance.services.*;
+import edu.ksu.canvas.model.Progress;
+import edu.ksu.canvas.oauth.OauthToken;
+import edu.ksu.canvas.requestOptions.MultipleSubmissionsOptions;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+@Scope(value="session")
+public class AssignmentSubmitter {
+    private static final Logger LOG = Logger.getLogger(AssignmentSubmitter.class);
+
+    @Autowired
+    private AttendanceAssignmentService assignmentService;
+
+    @Autowired
+    private AttendanceSectionService sectionService;
+
+    @Autowired
+    private AttendanceCourseService attendanceCourseService;
+
+    @Autowired
+    private AttendanceService attendanceService;
+
+    @Autowired
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    @Autowired
+    private AssignmentValidator assignmentValidator;
+
+    @Autowired
+    private CanvasAssignmentAssistant canvasAssignmentAssistant;
+
+
+    /**
+     * This function takes the summary for sections and one by one validates the attendance assignment and
+     * the canvas assignment. Then pushes all the grades and comments to the canvas assignment associated.
+     * @param isSimpleAttendance
+     * @param summaryForSections
+     */
+    public void submitCourseAttendances(boolean isSimpleAttendance, List<AttendanceSummaryModel> summaryForSections, Long courseId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)